### PR TITLE
SigmaProp type added 

### LINKED ISSUE #43: Implement Sigma type
Subtasks:
- [ ] refactoring of the interpreter to handle Sigma values

### PR DESCRIPTION
**Pull request by ergomorphic**
Close #43, #44 
This allows to pass ProveDlog and DHT values as context variables and in box registers and have access to them from scripts. (before this scripts didn't support this)
In particular current implementation of CrowdFunding implements passing publicImage as constants embedded in the script body, not as parameters.

### CODE REVIEW

File: src/main/scala/org/ergoplatform/ErgoLikeContext.scala [hunk 1]
```diff
@@ -25,6 +25,9 @@ class ErgoLikeContext(val currentHeight: Height,
 
   def withTransaction(newSpendingTransaction: ErgoLikeTransactionTemplate[_ <: UnsignedInput]): ErgoLikeContext =
     ErgoLikeContext(currentHeight, lastBlockUtxoRoot, boxesToSpend, newSpendingTransaction, self, extension)
+
+  def withSelf(newSelf: ErgoBox): ErgoLikeContext =
```
Comment (kushti) [inline src/main/scala/org/ergoplatform/ErgoLikeContext.scala]: Why can we have need for this?

Comment (ergomorphic) [reply src/main/scala/org/ergoplatform/ErgoLikeContext.scala]: @kushti I added it in some intermediate version, now it is not used. Remove?

**Comment by ergomorphic** [comment]
@kushti 
1) renamed Proof -> SigmaProp type
2) all existing examples remain unchanged, i.e. SigmaProp is not necessary to use in most cases
3) It solves the problem with passing ProveDlog values via context vars and registers is the scenario require this.
4) test cases are provided

File: build.sbt [hunk 2]
```diff
@@ -45,6 +45,7 @@ libraryDependencies ++= Seq(
 
 publishMavenStyle := true
 
+parallelExecution in Test := false
```
[the remark this answers is not in the document]
Comment (ergomorphic) [reply build.sbt]: @greenhat Completely and quite repeatedly. And I also recalled your fix, but it doesn't solve the problem, unfortunately. Also I don't like switching from SCollection to SCollectionType, just to make SBT happy. So we need more principled solution.

**Comment by kushti** [comment]
@ergomorphic approved, but please address my comment

**Pull request merged**